Sources/Build/BuildPlan.swift

```diff
@@ -628,11 +628,27 @@ public final class SwiftTargetBuildDescription {
         return (target.underlyingTarget as! SwiftTarget).swiftVersion
     }
 
-    /// If this target is a test target.
-    public let isTestTarget: Bool
+    /// Describes the purpose of a test target, including any special roles such as containing a list of discovered tests or
+    /// serving as the manifest target which contains the main entry point.
+    public enum TestTargetRole {
+        /// An ordinary test target, defined explicitly in a package, containing test code.
+        case `default`
+
+        /// A test target which was synthesized automatically, containing a list of discovered tests
+        /// from `plain` test targets.
+        case discovery
+
+        /// A test target which was synthesized automatically, containing a manifest file which was either provided
+        /// explicitly in the package, specified via a CLI flag, or else synthesized automatically to run all discovered tests.
+        case manifest(isSynthesized: Bool)
+    }
```

Comment:
cool 😎 

one thing I always disliked in the current code base was the term "test manifest" as it always confuses the readers with the package manifest. we can consider taking the opportunity and renaming this to "test driver" or "test entry point" or something along these lines. thoughts? 

cc @abertelrud @elsh @neonichu

Reply:
"test entry point" 👍

Reply:
I've done this! It ended up touching quite a lot of files, but I think it's a good improvement overall